File: examples/django/app/sms/providers/sms_trap.py

import requests
from django.conf import settings
from .base import SmsProvider
import logging


class SMSTrap(SmsProvider):
    def __init__(self, api_key, sender_id):
        self.base_url = settings.SMS_TRAP_BASE_URL
        # api_key and sender_id are not used in SMSTrap, but we're accepting them for demonstration
        self.api_key = api_key
        self.sender_id = sender_id

    def send(self, phones: list[str], message: str, additional: dict) -> dict:

        label = additional.get('label', 'transactional')

        params = []

        for phone in phones:
            params.append(('phones[]', phone))

        params.extend([
            ('message', message),
            ('label', label)
        ])

        response = requests.get(self.base_url, params=params)
        logging.debug(response)
        return response.json()

File: examples/laravel/app/Services/sms/SMSTrap.php
<?php

namespace App\Services\sms;

use Illuminate\Support\Facades\Http;

class SMSTrap implements SMS
{
    protected string $baseUrl = 'http://localhost:8080/api/v1/trap';

    public function send(array $phones, string $message, array $params): mixed
    {
        $query = [
            'message' => $message,
            'label' => $data['label'] ?? 'transactional',

        ];

        foreach ($phones as $phone) {
            $query['phones[]'] = $phone;
        }

        $response = Http::withOptions(['debug' => true])->get($this->baseUrl, $query);

        return $response->json();
    }
}
